sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -123,6 +125,31 @@ abstract class QueryTest extends PlanTest {
              |""".stripMargin)
     }
 
+    import TreeNodeJsonFormatter._
+    val logicalPlan = analyzedDF.queryExecution.analyzed
+    var logicalRDDs = logicalPlan.collect { case l: LogicalRDD => l }
+    var logicalRelations = logicalPlan.collect { case l: LogicalRelation => l }
+    try {
+      val jsonBackPlan = fromJSON(toJSON(logicalPlan)).asInstanceOf[LogicalPlan]
+      val normalized = jsonBackPlan transformDown {
```

Comment:
Can you comment why you need to do this? (i.e. RDDs/data are serializable to JSON)